WordPress/gutenberg
# [RNMobile] Refactor Caption component

`Gutenberg-mobile` PR -> https://github.com/wordpress-mobile/gutenberg-mobile/pull/1678

## Description
Our current `Caption` component implementation was meant to be within a block level component, but now there are cases where we just want to use the Caption UI like in the Gallery block.

So this PR adds a new component `BlockCaption` that will have the current logic wrapping a `Caption` component which will only have the UI.

## How has this been tested?
- Open the app
- Add an Image block
  - Set a caption
  - **Expect**: caption being set correctly
- Add a Video block
  - Set a caption
  - **Expect**: caption being set correctly
- Add a Gallery block
  - Add some images
    - Set captions
    - **Expect**: captions being set correctly
  - Set a caption to the Gallery
  - **Expect**: caption being set correctly 
- Save post
- Open it
  - **Expect**: captions are showing correctly

## Types of changes
Refactor

## Checklist:
